### ISSUE TITLE
Forest replicas created on hosts not specified in mlDatabaseHosts (using ml-gradle 3.2.0)

### ISSUE THREAD

**Issue by derms** [marklogic-community/ml-gradle#279]
Created a cluster with 2 enodes (enode1,enode2) and 3 dnodes (dnode1,dnode2,dnode3)

When using the deployment config below, the primary forests are correctly created on the three dnodes, but forest 'my-app-content-3-replica-1' was created on enode1.

```
mlContentForestsPerHost=1
mlDatabaseNamesAndReplicaCounts=my-app-content,1
mlDatabaseHosts=my-app-content,dnode1|dnode2|dnode3
```

Can you make it so that the replicas are only setup on the hosts defined by mlDatabaseHosts?

**Comment by rjrudin** [comment]
Doh! Didn't even think of replicas. This will go into 3.3.0, possibly 3.2.1 as well if I make that.

**Comment by rjrudin** [comment]
@derms I published version 3.2.0-216 of ml-app-deployer, which I believe fixes the problem. Also made a couple improvements to how mlDatabaseHosts works - there's additional logging, and the order of the hosts is preserved (was using HashMap before, now using LinkedHashMap).

Can you try this version out? You can do by via the following:

    buildscript {
        dependencies {
                classpath "com.marklogic:ml-app-deployer:3.2.0-216"
        }
    }
    plugins {
        id "com.marklogic.ml-gradle" version "3.2.0"
    }

**Comment by derms** [comment]
@rjrudin I confirm that it works now. Thanks!

**Comment by rjrudin** [comment]
Okay - you can use that for the time being, I'll try to get 3.3.0 in the next week or so.

**Comment by rjrudin** [comment]
This is fixed in ml-app-deployer verison 3.2.0-216 and 3.2.0-217.

**Comment by derms** [comment]
@rjrudin ok create thanks
